frontend/src/pages/Schedule/index.jsx

```diff
@@ -1,66 +1,66 @@
-                    <div className="flex flex-col mr-[10%] text-xl">
-                        <label for="groupSelect">Select a group:</label>
-                        <select id="groupSelect" onChange={(e) => setSelectedGroupId(e.target.value)} className="bg-zinc-300 rounded-md p-2 cursor-pointer">
-                            <option selected disabled hidden>Choose here</option>
+            <div id="divider" className="flex h-[92%] mr-3 ml-3 ">
+                <div className="flex flex-col w-[25%] h-full pt-4">
+                    <div className="flex flex-row w-full h-[5%] items-center ">
+                        <motion.button 
+                            whileHover={{ 
+                                scale: 1.1, 
+                                border: "2px solid", 
+                                transition: {
+                                    duration: 0.2,
+                                    ease: "easeInOut",
+                                }, 
+                            }}
+                            whileTap={{ scale: 0.9 }}
+                            onClick={() => {selectedGroupId !== null && setShowPortal(true)}} 
+                            className="bg-secondaryLight rounded-md h-[80%] pr-4 pl-4">
+                            + Add rule
+                        </motion.button>
+                        <motion.select
+                            whileHover={{ border: "2px solid" }}
+                            whileTap={{ border: "2px solid" }}
+                            onChange={(e) => setSelectedGroupId(e.target.value - 1)} 
+                            className="ml-auto mr-5 bg-secondaryLight rounded-md h-[80%] pr-3 pl-3 cursor-pointer">
+                            <option selected disabled hidden>Group</option>
                             {groups.length !== 0 && groups.map((group) => 
                                 <option value={group.id}>{group.name}</option>
                             )}
-                        </select>
+                        </motion.select>
                     </div>
-                    <div className="flex flex-col text-xl">
-                        <label for="templateSelect">Select a template:</label>
-                        <select id="templateSelect" onChange={(e) => setSelectedTemplateId(e.target.value)} className="bg-zinc-300 rounded-md p-2 cursor-pointer">
-                            <option selected disabled hidden>Choose here</option>
-                            {templates.length !== 0 && templates.map((template) => 
-                                <option value={template.id}>{template.name}</option>
-                            )}
-                        </select>
+                    {showPortal && <ScheduleModal
+                            showPortal={showPortal} 
+                            setShowPortal={setShowPortal}
+                            selectedGroup={groups.at(selectedGroupId)} />
+                    }
```

Comment:
tens o boolean aqui fora e la dentro nao seria necessario so la dentro

Reply:
Penso que somente tendo isso lá dentro força o load do modal, por isso irei tirar o que está lá dentro e irei deixar o de fora. Mas vou testar isso.